cdap-app-templates/cdap-etl/cdap-etl-core/src/test/java/co/cask/cdap/etl/planner/ConnectorDagTest.java

```diff
@@ -650,4 +652,66 @@ public void testIsolate() {
     Assert.assertEquals(expected, cdag);
   }
 
+  @Test
+  public void testConditionDag() throws Exception {
+
+    /*
+          file - csv - c1 - t1---agg1--agg2---sink1
+                        |
+                        ----c2 - sink2
+                             |
+                              ------c3 - sink3
+
+     */
+
+    ConnectorDag dag = ConnectorDag.builder()
+      .addConnection("file", "csv")
+      .addConnection("csv", "c1")
+      .addConnection("c1", "t1")
+      .addConnection("t1", "agg1")
+      .addConnection("agg1", "agg2")
+      .addConnection("agg2", "sink1")
+      .addConnection("c1", "c2")
+      .addConnection("c2", "sink2")
+      .addConnection("c2", "c3")
+      .addConnection("c3", "sink3")
+      .addReduceNodes("agg1", "agg2")
+      .build();
+
+    dag.insertConnectors();
+    List<Dag> actual = dag.splitByConditions(new HashSet<>(Arrays.asList("c1", "c2", "c3")));
+
+    Dag dag1 = new Dag(
+      ImmutableSet.of(
+        new Connection("file", "csv"),
+        new Connection("csv", "c1")));
+    Dag dag2 = new Dag(
+      ImmutableSet.of(
+        new Connection("c1", "t1"),
+        new Connection("t1", "agg1.connector")));
```

Comment:
I see. The extra connector is getting added by the `insertConnector` method which we call before splitting by condition nodes. Perhaps its better to split first based on the condition nodes and then for individual subdags do `insertConnector`?